modules/libcom/src/osi/os/WIN32/osdEnv.c

```diff
@@ -34,25 +34,8 @@
  */
 LIBCOM_API void epicsStdCall epicsEnvSet (const char *name, const char *value)
 {
-    char *cp;
-
     iocshEnvClear(name);
-
-    cp = mallocMustSucceed (strlen (name) + strlen (value) + 2, "epicsEnvSet");
-    strcpy (cp, name);
-    strcat (cp, "=");
-    strcat (cp, value);
-    if (putenv (cp) < 0) {
-        errPrintf(
-                -1L,
-                __FILE__,
-                __LINE__,
-                "Failed to set environment parameter \"%s\" to \"%s\": %s\n",
-                name,
-                value,
-                strerror (errno));
-        free (cp);
-    }
+    _putenv_s(name, value);
 }
```

Comment:
Do we need to check the return code,  or that value does not exceed `_MAX_ENV` ?

Reply:
Well, probably yes.  Although none of the other impls check for `ENOMEM` from `putenv()` or `setenv()`.

How should an error be communicated?  Change to a non-void return type?  (I hesitate to printf)

Reply:
Changing the return type feels like a lot of work might potentially need to be done to trace the error through, though maybe that is the most correct.  Do you feel using `errPrintf` creates a dependency between the osi and other layers? It is probably unlikely for `_putenv_s` to fail, but if it did could we end up with a hard to trace error? That was probably more my concern, though errors caused by corrupt/missing environment variables are probably clearer than others to identify, so maybe it is not such an issue.

Reply:
The code Freddie is deleting was already calling `mallocMustSucceed()` which would suspend and dump a stack trace if it failed, and `errPrintf()` if the `putenv()` failed. Sticking with the latter wouldn't make matters any worse, so I say stick with that.